__tests__/View-tests/components/challenge/challengeGuest.test.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react-native";
+import { Challenge } from "@/src/views/components/posts/challenge";
+import FirestoreCtrl from "@/src/models/firebase/FirestoreCtrl";
+
+jest.mock("@/src/models/firebase/FirestoreCtrl");
+
+describe("Challenge Component - Guest User Restrictions", () => {
+  const mockNavigation = { navigate: jest.fn() };
+  const mockFirestoreCtrl = new FirestoreCtrl();
+
+  const mockChallengeDB = {
+    challenge_id: "challenge-1",
+    uid: "user-1",
+    description: "Test Challenge Description",
+    image_id: "https://example.com/test-image.jpg",
+    likes: ["user-1"],
+    date: new Date(),
+    caption: "Test Challenge Caption",
+    challenge_description: "Test Challenge Description",
+  };
+
+  const mockCurrentUserGuest = {
+    name: "Guest",
+    uid: "guest-1",
+    email: "guest@gmail.com",
+    createdAt: new Date(),
+  };
+
+  it("does not allow guest users to like a challenge", () => {
```

Comment:
This is a logic testing, so it should not be in the View-tests, but more in the viewmodel tests